conversant/prompt_chatbot.py

```diff
@@ -118,6 +128,52 @@ def latest_prompt(self) -> str:
         """
         return self.prompt_history[-1]
 
+    def _update_max_context_examples(
+        self, prompt_size: int, max_context_examples: int
+    ) -> int:
+        """Adjust max_context_examples until a possible prompt size.
+
+        if this is not possible, send an error message.
+
+        Args:
+            prompt_size (int): Number of tokens of the prompt
+            max_context_examples (int): The length of the chat history for
+            the chatbot to use in reply.
+
+        Returns:
+           int: updated max_context_examples
+        """
+        # Store original values
+        original_size = prompt_size
+        # If the size of chat_history is smaller than max_context_examples
+        # the value of the variable is already updated with the size value
+        trimmed_max_examples = min(len(self.chat_history), max_context_examples)
+
+        # Reduce max_context_examples until the number of token of the prompt
+        # is less than maximum or reaches 1
+        for size in self.prompt_size_history[-max_context_examples:]:
+            prompt_size -= size
+            trimmed_max_examples -= 1
+            if prompt_size <= self.max_prompt_size:
+                warnings.warn(
+                    "The parameter max_context_examples was reduced for this turn, "
+                    f"from the desired config {max_context_examples} to "
+                    f"{trimmed_max_examples}, so that "
+                    f"the total amount of tokens does not exceed {MAX_GENERATE_TOKENS}."
+                )
+                return trimmed_max_examples
+
+        raise ValueError(
+            "The total number of tokens (prompt and prediction) cannot exceed "
+            f"{MAX_GENERATE_TOKENS}. Try using a shorter start prompt, sending "
+            "smaller text messages in the chat, or setting a smaller value "
+            "for the parameter max_tokens. More details: \n"
+            f" - Start Prompt: {self.start_prompt_size} tokens \n"
+            f" - Messages sent in chat: {original_size - self.start_prompt_size} "
+            f" tokens \n - Parameter max_tokens: {self.client_config['max_tokens']}"
+            " tokens"
```

Comment:
```suggestion
            "The total number of tokens (prompt and prediction) cannot exceed "
            f"{MAX_GENERATE_TOKENS}. Try using a shorter start prompt, sending "
            "smaller text messages in the chat, or setting a smaller value "
            "for the parameter max_tokens. More details:\n"
            f" - Start Prompt: {self.start_prompt_size} tokens\n"
            f" - Messages sent in chat: {original_size - self.start_prompt_size} "
            f"tokens\n - Parameter max_tokens: {self.client_config['max_tokens']} "
            "tokens"
```
nit: extra whitespaces